wagtail/models/reference_index.py

```diff
@@ -308,8 +308,25 @@ def _extract_references_from_object(cls, object):
             content_path (str): The path to the piece of content on the source
                                 object instance where the reference was found
         """
-        # Extract references from fields
-        for field in object._meta.get_fields():
+        for field in object._meta.get_fields(include_hidden=True):
+            # New: Process custom extract_references first
```

Comment:
```suggestion
            # First - Extract custom extract_references from fields
```

Instead of 'new', as this looks like a PR only comment.